File: python/model/guerrero.py
from model.personaje import Personaje

# Herencia
class Guerrero(Personaje):

    # Constructor
    def __init__(self, id, nombre, nivel):

        # llama constructor padre
        super().__init__(id, nombre, nivel)

    # sobrescritura
    def habilidad_especial(self):

        print("Ataque con espada")

File: java/src/model/Guerrero.java
package model;

// Herencia
public class Guerrero extends Personaje {

    // Constructor
    public Guerrero(int id, String nombre, int nivel) {

        // super llama al constructor padre
        super(id, nombre, nivel);
    }

    // Sobrescritura
    @Override
    public void habilidadEspecial() {

        System.out.println("Ataque con espada");
    }
}